packages/astro/src/core/ssr/index.ts

```diff
@@ -245,15 +247,23 @@ export async function render(renderers: Renderer[], mod: ComponentInstance, ssrO
 
   // inject CSS
   [...getStylesForURL(filePath, viteServer)].forEach((href) => {
-    tags.push({
-      tag: 'link',
-      attrs: {
-        rel: 'stylesheet',
-        href,
-        'data-astro-injected': true,
-      },
-      injectTo: 'head',
-    });
+    if (mode === 'development' && svelteAndVueStylesRE.test(href)) {
```

Comment:
Only runs in dev; only runs if it matches the Svelte/Vue style pattern of `.(vue|svelte)?(vue|svelte)&type=style&lang.css`. Ordering of the search params should be set in stone (i.e. they will appear in this order)